docs/_getting-started/dev-mode.md

```diff
@@ -1,64 +1,341 @@
+
+```
+
+The transaction hash can be used to retrieve the transaction details using the Geth Javascript console, which will return the following information:
+
+```terminal
+
+{
+  accessList: [],
+  blockHash: "0xc076c88200618f4cbbfb4fe7c3eb8d93566724755acc6c4e9a355cc090eeee77",
+  blockNumber: 3,
+  chainId: "0x539",
+  from: "0x7aa16266ba3d309e3cb278b452b1a6307e52fb62",
+  gas: 43724,
+  gasPrice: 3172359839,
+  hash: "0xfa3cd8df6841c5d3706d3bacfb881d2b985d0b55bdba440f1fdafa4ed5b5cc31",
+  input: "0x6057361d0000000000000000000000000000000000000000000000000000000000000038",
+  maxFeePerGas: 4032048134,
+  maxPriorityFeePerGas: 2500000000,
+  nonce: 2,
+  r: "0x859b88062715c5d66b9a188886ad51b68a1e4938d5932ce2dac874c104d2b26",
+  s: "0x61ef6bc454d5e6a76c414f133aeb6321197a61e263a3e270a16bd4a65d94da55",
+  to: "0x4aa11ddfd817dd70e9ff2a2bf9c0306e8ec450d3",
+  transactionIndex: 0,
+  type: "0x2",
+  v: "0x1",
+  value: 0
+}
+
+```
+
+The `from` address is the account that sent the transaction, the `to` address is the deployment address of the contract. The value entered into Remix is now in storage at that contract address. This can be retrieved using Remix by calling the `retrieve` function - to do this simpyl click the `retrieve` button. Alternatively, it can be retrieved using `web3.getStorageAt` using the Geth Javascript console. The following command returns the value in the contract storage (replace the given address with the correct one displayed in the Geth logs).
+
+```shell
+
+web3.eth.getStorageAt("0x407d73d8a49eeb85d32cf465507dd71d507100c1", 0)
+
+```
+
+This returns a value that looks like the following:
+
+```terminal
+
+"0x000000000000000000000000000000000000000000000000000000000000000038"
+
+```
+
+
+The returned value is a left-padded hexadecimal value. For example, the return value `0x000000000000000000000000000000000000000000000000000000000000000038` corresponds to a value of `56` entered as a uint256 to Remix. After converting from hexadecimal string to decimal number the returned value should be equal to that provided to Remix in the previous step.
+
+## Reusing --datadir
+
+This tutorial used an ephemeral blockchain that is completely destroyed and started afresh during each dev-mode session. However, it is also possible to create a persistent blockchain that can be restarted across multiple sessions. This is done by providing the `--datadir` flag and a directory name when starting Geth in dev-mode. However, if there is already an account in Geth's keystore, Geth will not start because it cannot automatically unlock that existing account. To resolve this issue, the password defined when the account was created can be saved to a text file and its path passed to the `--password` flag on starting Geth, for example if `password.txt` is saved in the top-level `go-ethereum` directory:
+
+```shell
+
+geth --datadir dev-chain --dev --http --http.api personal,web3,eth,net --http.corsdomain "remix.ethereum.org" --password password.txt
```

Comment:
Let's handle the case with `--password` separately. There are two things here:

- It is possible to reuse the data directory. We need to one paragraph + example for this.

- When specifying `--datadir` together with `--dev`, not only the blockchain data will persist, but also the 'developer account' (i.e. the account which is created + unlocked by `--dev`).

- It is even possible to create the developer account yourself, or import a pre-existing key into the keystore. However, if you do this, `--dev` cannot unlock the account automatically, so it will fail to start. It's important to realize that this is a corner case which will only happen for manual key creation or import. To make `--dev` work in this specific case, the password must be given using `--password`. This should be documented as a separate example.